api/routes/catalog.py

```diff
@@ -19,7 +19,7 @@ async def create_dataset(document: DatasetMetadataDOC, user: Annotated[User, Dep
     submission = document.as_submission()
     await submission.insert()
```

Comment:
I believe we don't need to do the submission insert (See "Write" section of https://beanie-odm.dev/tutorial/relations/)

Reply:
Yeah looks like we don't need it. Fixed it.